### PR TITLE
adding new techniques (part 2)

### PR DESCRIPTION
**Pull request by lzehl**

### CODE REVIEW

**Comment by lzehl** [comment]
@UlrikeS91 this is ready now for a review :)

**Review by apdavison — COMMENTED** [review]
Do you also plan to add "small-scale simulation" and "medium-scale simulation"? How will the line be drawn?

How about just "simulation"? Or "simulation" + "HPC simulation" if you think the distinction of scale is important.

**Comment by lzehl** [comment]
@apdavison We can yes. For now I was just focusing on terms that were already present in some way in the old system. We need to go as deep as it would make sense for queries. I think large/medium/small makes sense. is large-scale not a synonym to HPC? or are there large-scale simulations that can run without an HPC?

File: instances/technique/nonrigidMotionCorrection.jsonld [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  "@context": {
+    "@vocab": "https://openminds.ebrains.eu/vocab/"
+  },
+  "@id": "https://openminds.ebrains.eu/instances/technique/nonrigidMotionCorrection",
+  "@type": "https://openminds.ebrains.eu/controlledTerms/Technique",
+  "definition": null,
+  "description": null,
+  "interlexIdentifier": null,
+  "knowledgeSpaceLink": null,
+  "name": "nonrigid motion correction",
```
Comment (UlrikeS91) [inline instances/technique/nonrigidMotionCorrection.jsonld:11]: This is lacking the "-" between "non" and "rigid" (or a space, but I think that would be weird). The `@id` need to be adjusted accordingly.

File: instances/technique/nonrigidRegistration.jsonld [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "@context": {
+    "@vocab": "https://openminds.ebrains.eu/vocab/"
+  },
+  "@id": "https://openminds.ebrains.eu/instances/technique/nonrigidRegistration",
+  "@type": "https://openminds.ebrains.eu/controlledTerms/Technique",
+  "definition": null,
+  "description": null,
+  "interlexIdentifier": null,
+  "knowledgeSpaceLink": null,
+  "name": "nonregid registration",
```
Comment (UlrikeS91) [inline instances/technique/nonrigidRegistration.jsonld]: same comment as above; additionally, "rigid" has a typo ("r**E**gid")

File: instances/technique/rigidRegistration.jsonld [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  "@context": {
+    "@vocab": "https://openminds.ebrains.eu/vocab/"
+  },
+  "@id": "https://openminds.ebrains.eu/instances/technique/rigidRegistration",
+  "@type": "https://openminds.ebrains.eu/controlledTerms/Technique",
+  "definition": null,
+  "description": null,
+  "interlexIdentifier": null,
+  "knowledgeSpaceLink": null,
+  "name": "regid registration",
```
Comment (UlrikeS91) [inline instances/technique/rigidRegistration.jsonld]: "rigid" with typo again ("rEgid")

**Review by UlrikeS91 — CHANGES_REQUESTED** [review]

**Comment by apdavison** [comment]
Hi Lyuba, I think "small", "medium" and "large" are problematic, because there is no agreed upon definition for how to define the scale of a simulation (number of neurons, number of synapses, volume of tissue simulated, number of processor-hours required...?), and for sure the boundaries change with time (what was large-scale 10 years ago is now medium- or small-scale).

I suggested "HPC simulation" as a more concrete, well-defined term that is approximately equivalent to the intent of "large-scale", but really I would be inclined to just have "simulation".

File: instances/technique/nonrigidMotionCorrection.jsonld (hunk 1 above)
Comment (lzehl) [reply instances/technique/nonrigidMotionCorrection.jsonld:11]: https://www.merriam-webster.com/dictionary/nonrigid should be correct

File: instances/technique/nonrigidRegistration.jsonld (hunk 2 above)
Comment (lzehl) [reply instances/technique/nonrigidRegistration.jsonld]: cf above

File: instances/technique/rigidRegistration.jsonld (hunk 3 above)
Comment (lzehl) [reply instances/technique/rigidRegistration.jsonld]: cf above

**Comment by lzehl** [comment]
@apdavison just "simulation" seems too vague for me to be of real use. I've entered "large-scale simulation" because it has a meaning to people from the field (discussed it with some NEST people), but HPCsimulation would make them also happy (I guess). Could you have a look at https://search.kg.ebrains.eu/instances/069e4718-2b4a-4056-9c8f-e646a841b2dd and tell me which controlled techniques would you filter out of the given methods? (I've kept now "communication profiling", "large-scale simulation", "quantitative analysis" and "qualitative analysis")

**Comment by UlrikeS91** [comment]
> @apdavison just "simulation" seems too vague for me to be of real use. I've entered "large-scale simulation" because it has a meaning to people from the field (discussed it with some NEST people), but HPCsimulation would make them also happy (I guess). Could you have a look at https://search.kg.ebrains.eu/instances/069e4718-2b4a-4056-9c8f-e646a841b2dd and tell me which controlled techniques would you filter out of the given methods? (I've kept now "communication profiling", "large-scale simulation", "quantitative analysis" and "qualitative analysis")

The PR is basically ready, but @lzehl and I would like to wait for @apdavison feedback about the question around the "simulation" technique. @apdavison would you be able to have a look and tell us which techniques would be suitable?

**Comment by apdavison** [comment]
Having now reviewed the level of detail in the other "technique" instances, I definitely think that plain "simulation" would be best. If you want more specificity, then I would put both "HPC simulation" and "neuromorphic simulation" for that specific dfataset. "Large-scale" should be avoided, as the term is subjective and doesn't have a stable meaning across the field or across time (i.e. different people have a different idea about what it means, and people's idea of its meaning changes as computing technology improves).

I would be inclined to remove "communication profiling", as this is a technique from computer engineering, not from neuroscience.

**Comment by lzehl** [comment]
@apdavison and @UlrikeS91 I've renamed "large-scale simulation" to "HPC simulation", and added "simulation" and "neuromorphic simulation" as well. I will leave "communication profiling" because it is still a technique. Note that what we register under techniques is not purely neuroscience, but all techniques that we need to mention for the data we register in the KG.

**Comment by lzehl** [comment]
I've also updated the image registration techniques and the transformation techniques according to the feedback I received from Timo.

File: instances/technique/imageRegistration.jsonld [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "@context": {
+    "@vocab": "https://openminds.ebrains.eu/vocab/"
+  },
+  "@id": "https://openminds.ebrains.eu/instances/technique/registration",
+  "@type": "https://openminds.ebrains.eu/controlledTerms/Technique",
+  "definition": "An 'image registration' is a process of bringing a set of images into the same coordinate system.",
+  "description": null,
+  "interlexIdentifier": null,
+  "knowledgeSpaceLink": null,
+  "name": "transformation",
```
Comment (UlrikeS91) [inline instances/technique/imageRegistration.jsonld]: `@id` vs. `name`: registration or transformation? Definition points to registration, but also says "image registration" which is yet another thing.

**Review by UlrikeS91 — CHANGES_REQUESTED** [review]

File: instances/technique/imageRegistration.jsonld (hunk 4 above)
Comment (lzehl) [reply instances/technique/imageRegistration.jsonld]: screw up on my side... should be corrected now

**Review by UlrikeS91 — APPROVED** [review]

**Pull request merged**